src/oscar/apps/basket/views.py

```diff
@@ -54,10 +54,16 @@ def get_shipping_methods(self, basket):
             basket=self.request.basket, user=self.request.user,
             request=self.request)
 
+    def get_default_shipping_address(self):
+        if self.request.user.is_authenticated():
+            shipping_addresses = self.request.user.addresses.filter(is_default_for_shipping=True)
```

Comment:
Can do this all in one query: `user.addresses.filter(is_default_for_shipping=True).first()`, which will be `None` if none exists.